pandas/core/arrays/string_.py

```diff
@@ -85,6 +85,22 @@ def construct_array_type(cls) -> "Type[StringArray]":
     def __repr__(self) -> str:
         return "StringDtype"
 
+    def __from_arrow__(self, array):
+        """Construct StringArray from passed pyarrow Array"""
+        import pyarrow
+
+        if isinstance(array, pyarrow.Array):
+            chunks = [array]
+        else:
+            chunks = array.chunks
+
+        results = []
+        for arr in chunks:
+            str_arr = StringArray(np.array(arr))
```

Comment:
@TomAugspurger Is there a way to turn off validation when creating a StringArray? (in this case I know the elements will be strings or None) 
I didn't directly see a private method on StringArray that does this

Reply:
Not at the moment. In `StringArray.__init__` we have

```python
skip_validation = isinstance(values, type(self))
```

I'd be open to a public API for allowing the user / library to state that they have the right values.

One slight concern though: right now it's expect that the values are strings or `nan`, and we may expect to change the NA value to change in the future. Libraries will have to adapt to that if they're claiming to be compatible.

Reply:
Ah, so actually the validation right now *is* needed here, even if it is just to convert None to nan?

Reply:
I think the None -> nan replacement happens in in `_from_sequence`, though I have a TODO to move it to `_validate` since that's already requires a pass over the data.

Reply:
Ah, good catch, I switched to `_from_sequence` for now, and added a test for it.